FirebaseAuth/Sources/Swift/Auth/Auth.swift

```diff
@@ -76,56 +74,51 @@ import FirebaseCoreExtension
 
 @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
 extension Auth: AuthInterop {
+  func getTokenInternal(forcingRefresh forceRefresh: Bool) {
+    // Enable token auto-refresh if not already enabled.
+    if !autoRefreshTokens {
+      AuthLog.logInfo(code: "I-AUT000002", message: "Token auto-refresh enabled.")
+      autoRefreshTokens = true
+      scheduleAutoTokenRefresh()
+
+      #if canImport(UIKit) // Is a similar mechanism needed on macOS?
+        applicationDidBecomeActiveObserver =
+          NotificationCenter.default.addObserver(
+            forName: UIApplication.didBecomeActiveNotification,
```

Comment:
I'm wondering what happen with app extensions, in this way I think UIApplication.didBecomeActiveNotification and didEnterBackgroundNotification would be not fired @paulb777

Reply:
Good question. This implementation should match the current Objective-C one. Do you see an issue?

Reply:
Sure, even the didBecomeActiveNotification is available on extensions and the other one as well, it won't be fired. so I would check if it is an extension or not using for example the following code:

```
var isRunningInExtension: Bool {
    // Check if the bundle path ends with `.appex` (for iOS)
    let bundlePath = Bundle.main.bundlePath
    let isBundlePathExtension = bundlePath.hasSuffix(".appex")
    
    // Check if the `NSExtension` key exists in the `Info.plist` 
    let isNSExtensionPresent = (Bundle.main.infoDictionary?["NSExtension"] as? [String: Any]) != nil
    
    return isBundlePathExtension || isNSExtensionPresent
}
```

And In case of being true (the output) try to make a check in other way. Dispatch Queues with async after a time don't work on certain extensions as well as timers ... (Because apple timers and dispatch asyncs tends to sleep when the device is locked or with no activity for a long time) So you can take another approach like going with a counter with the clock time with a sleep or infinite loop. They are not good approaches but extensions are very restricted.